From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 11 Mar 2022 17:39:21 -0500
Subject: [PATCH] Switch to use find_package for codec2, link with codec2
 target.

Codec2 breaks standard and calls its import library "libcodec2.lib",
which doesn't work with the Unix-style pkg-config file that tells
programs to link with "codec2"=="codec2.lib". Codec2 has been providing
its own codec2-config.cmake file for a while now, so just use it to make
things simpler.
---
 CMakeLists.txt       | 3 +--
 apps/CMakeLists.txt  | 4 ++--
 tests/CMakeLists.txt | 1 -
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 791243b..37a7b52 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,10 +24,9 @@ message(STATUS "# Checking dependencies")
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 
-include(FindPkgConfig)
 include(GNUInstallDirs)
 
-pkg_check_modules(CODEC2 REQUIRED codec2)
+find_package(codec2 REQUIRED)
 set(Boost_USE_STATIC_LIBS FALSE)
 find_package(Boost COMPONENTS program_options REQUIRED)
 
diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index a419e7c..1d80242 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -1,7 +1,7 @@
 add_executable(m17-demod m17-demod.cpp)
-target_link_libraries(m17-demod PRIVATE m17cxx ${CODEC2_LIBRARIES} ${Boost_LIBRARIES})
+target_link_libraries(m17-demod PRIVATE m17cxx codec2 ${Boost_LIBRARIES})
 
 add_executable(m17-mod m17-mod.cpp)
-target_link_libraries(m17-mod PRIVATE m17cxx ${CODEC2_LIBRARIES} ${Boost_LIBRARIES} Threads::Threads)
+target_link_libraries(m17-mod PRIVATE m17cxx codec2 ${Boost_LIBRARIES} Threads::Threads)
 
 install(TARGETS m17-demod m17-mod RUNTIME DESTINATION bin)
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index d188b05..3937534 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,5 +1,4 @@
 include(GoogleTest)
-pkg_check_modules(CODEC2 REQUIRED codec2)
 
 include_directories (
     ${TEST_SOURCE_DIR}
-- 
2.35.1

